abort-pending-builds-based-on-string-searched-on-logs.groovy

```diff
@@ -0,0 +1,38 @@
+findItems(jenkins.model.Jenkins.instance.items.findAll());
+
+def findItems(items) {
+  for (item in items) {
+    switch(item) {
+        case org.jenkinsci.plugins.workflow.multibranch.WorkflowMultiBranchProject:
+            item.getItems()?.each { itemMultibranch ->
+                searchStringInLogBuild(itemMultibranch, "String that you are looking for")
+            }
+            break;
+        case org.jenkinsci.plugins.workflow.job.WorkflowJob:
+            searchStringInLogBuild(item, "String that you are looking for")
+            break;
+
+        // you can add other kinds of Items using a case clause, or even a default value.
+
+        case com.cloudbees.hudson.plugins.folder.Folder:
+            findItems(item.getItems());
+            break;
+    }
+  }
+}
+
+def searchStringInLogBuild(item, string) {
+  item.getBuilds()?.each { build ->
+    if (build.isBuilding()) {
```

Comment:
If we are looking for running builds, we should probably search them from the busy Executor list as in https://github.com/cloudbees/jenkins-scripts/blob/424ac777961cb29e14dbdfdd5a3d9ef65d69cb10/ProperlyStopOnlyRunningPipelines.groovy#L45-L48. This is much more efficient and less resource consuming than the `allItems` search.

Reply:
Thanks for the suggestion! Really useful